src/openrct2-ui/windows/RideList.cpp

```diff
@@ -172,872 +143,795 @@ static constexpr const rct_string_id page_names[] = {
-                            break;
-
-                        window_bubble_list_item(w, current_list_position);
+                        ft.Add<uint16_t>(ridePtr->guests_favourite);
+                        formatSecondary = ridePtr->guests_favourite == 1 ? STR_GUESTS_FAVOURITE_LABEL
+                                                                         : STR_GUESTS_FAVOURITE_PLURAL_LABEL;
                     }
-                }
-                break;
+                    break;
+            }
+
+            if (formatSecondaryEnabled)
+            {
+                ft.Rewind();
+                ft.Add<rct_string_id>(formatSecondary);
+            }
+            DrawTextEllipsised(&dpi, { 160, y - 1 }, 157, format, ft);
+            y += SCROLLABLE_ROW_HEIGHT;
         }
+    }
 
-        list_index++;
+    void RefreshListWrapper()
+    {
+        RefreshList();
     }
 
-    w->selected_list_item = -1;
-    w->Invalidate();
-}
+private:
+    /**
+     *
+     *  rct2: 0x006B38EA
+     */
+    void DrawTabImages(rct_drawpixelinfo* dpi)
+    {
+        int32_t sprite_idx;
+
+        // Rides tab
+        sprite_idx = SPR_TAB_RIDE_0;
+        if (this->page == PAGE_RIDES)
+            sprite_idx += this->frame_no / 4;
+        gfx_draw_sprite(
+            dpi, ImageId(sprite_idx),
+            this->windowPos + ScreenCoordsXY{ this->widgets[WIDX_TAB_1].left, this->widgets[WIDX_TAB_1].top });
+
+        // Shops and stalls tab
+        sprite_idx = SPR_TAB_SHOPS_AND_STALLS_0;
+        if (this->page == PAGE_SHOPS_AND_STALLS)
+            sprite_idx += this->frame_no / 4;
+        gfx_draw_sprite(
+            dpi, ImageId(sprite_idx),
+            this->windowPos + ScreenCoordsXY{ this->widgets[WIDX_TAB_2].left, this->widgets[WIDX_TAB_2].top });
+
+        // Information kiosks and facilities tab
+        sprite_idx = SPR_TAB_KIOSKS_AND_FACILITIES_0;
+        if (this->page == PAGE_KIOSKS_AND_FACILITIES)
+            sprite_idx += (this->frame_no / 4) % 8;
+        gfx_draw_sprite(
+            dpi, ImageId(sprite_idx),
+            this->windowPos + ScreenCoordsXY{ this->widgets[WIDX_TAB_3].left, this->widgets[WIDX_TAB_3].top });
+    }
 
-static void window_ride_list_close_all(rct_window* w)
-{
-    for (auto& ride : GetRideManager())
+    /**
+     * Bubbles an item one position up in the window list.  This is done by swapping
+     * the two locations.
+     *  rct2: New function not from rct2
+     */
+    void BubbleListItem(int32_t index)
     {
-        if (ride.status != RideStatus::Closed && ride.GetClassification() == static_cast<RideClassification>(w->page))
+        std::swap(_rideList[index], _rideList[index + 1]);
+    }
+
+    /**
+     * Used in RefreshList() to handle the sorting of the list.
+     * Uses a lambda function (predicate) as exit criteria for the algorithm.
+     */
+    template<typename TSortPred> void SortList(int32_t& currentListPosition, const Ride* thisRide, const TSortPred& pred)
```

Comment:
```suggestion
    template<typename TSortPred> int32_t SortList(int32_t currentListPosition, const Ride* thisRide, const TSortPred& pred)
```
Return the new index instead of passing it by argument reference.